src/controller/python/chip/interaction_model/delegate.py

```diff
@@ -32,21 +35,59 @@
     "CommandIndex" / Int8ul,
 )
 
+# AttributePath should not contain padding
+AttributePathStruct = Struct(
+    "EndpointId" / Int16ul,
+    "ClusterId" / Int32ul,
+    "AttributeId" / Int32ul,
+)
+
+
+@dataclass
+class AttributePath:
+    nodeId: int
+    endpointId: int
+    clusterId: int
+    attributeId: int
+
+    def __str__(self):
+        return "[NodeId: {}, EndpointId: {}, ClusterId: {}, AttributeId: {}]".format(self.nodeId, self.endpointId, self.clusterId, self.attributeId)
+
+    def __eq__(self, v):
```

Comment:
does `@dataclass` not provide its own eq that does the same thing?

Reply:
Yes, dataclass already provides the __eq__ and __repr__, so __str__ is also not required in this case.